r3f/src/ZestyBanner.js

```diff
@@ -45,13 +48,53 @@ export default function ZestyBanner(props) {
     }
   }, [bannerData]);
 
+
   const onClick = (event) => {
     const banner = bannerData;
     let url = banner.url || banner.properties?.url;
     openURL(url);
     if (props.beacon) sendOnClickMetric(props.adUnit, bannerData.campaignId);
   };
 
+  const checkVisibility = () => {
```

Comment:
Hmm any way we could stop repeating this function across multiple files? Would be great if we could consolidate.